### PR TITLE
IGAPP-831: Implemented different news preview lenghts

### PR DESCRIPTION
**Pull request by Cmd190**
The news preview length is now dependent from the viewport. 
Large Screen: 
image

Medium Screen:
image

Small Screen:
image

### CODE REVIEW

File: release-notes/2021.11.3/IGAPP-259-Add-contrast-for-setting-switches.yml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 issue_key: IGAPP-259
 show_in_stores: true
 platforms:
-  - native
+   - ios
+  - android
```
Comment (klinzo) [inline release-notes/2021.11.3/IGAPP-259-Add-contrast-for-setting-switches.yml:5]: indentation, same in other release note

File: release-notes/unreleased/IGAPP-831-news.preview-length.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+issue_key: IGAPP-259
```
Comment (klinzo) [inline release-notes/unreleased/IGAPP-831-news.preview-length.yml:1]:
```suggestion
issue_key: IGAPP-831
```

File: release-notes/unreleased/IGAPP-831-news.preview-length.yml [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+show_in_stores: true
+platforms:
+   - ios
+  - android
```
Comment (klinzo) [inline release-notes/unreleased/IGAPP-831-news.preview-length.yml:5]:
```suggestion
   - web
```

File: release-notes/unreleased/IGAPP-831-news.preview-length.yml (hunk 2 above)
Comment (klinzo) [reply release-notes/unreleased/IGAPP-831-news.preview-length.yml:1]: Please also replace the dot `.` with a `-` in the file name

File: web/src/components/Tooltip.tsx [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import React, { ReactElement, ReactNode, useCallback, useEffect, useState } from 'react'
 import styled, { css } from 'styled-components'
 
-import dimensions from '../constants/dimensions'
+import dimensions from '../constants/dimensions'
```
Comment (klinzo) [inline web/src/components/Tooltip.tsx:4]: Revert

File: web/src/constants/dimensions.ts [hunk 5]
```diff
@@ -6,13 +6,19 @@ export type DimensionsType = {
   toolbarWidth: number
   headerHeightLarge: number
   headerHeightSmall: number
+  smallViewportBorderValue : number
+  mediumViewportBorderValue : number
```
Comment (klinzo) [inline web/src/constants/dimensions.ts:10]:
```suggestion
  smallViewportThreshold : number
  mediumViewportThreshold : number
```

File: web/src/components/NewsListItem.tsx [hunk 6]
```diff
@@ -58,12 +61,23 @@ type PropsType = {
...
 
+  const maxChars = () => {
+    const width = window.innerWidth
+    if (width < dimensions.mediumViewportBorderValue && width >= dimensions.smallViewportBorderValue) {
+      return NUM_OF_CHARS_ALLOWED_MEDIUM_SCREEN
+    }
+
+    return width < dimensions.smallViewportBorderValue
+      ? NUM_OF_CHARS_ALLOWED_SMALL_SCREEN
+      : NUM_OF_CHARS_ALLOWED_LARGE_SCREEN
```
Comment (SteffiStuffel) [inline web/src/components/NewsListItem.tsx:72]: if think a `if..else if... else` would be much nicer than `if` and `?:` mixed.

```
if(width < small) {
  return SMALL_SCREEN
} else if (width < medium) {
  return MEDIUM_SCREEN
} else {
  return LARGE_SCREEN
}
```

Comment (klinzo) [inline web/src/components/NewsListItem.tsx:72]: If you change the order of the checks you can get rid of one check here

File: web/src/components/NewsListItem.tsx [hunk 7]
```diff
@@ -5,12 +5,15 @@ import styled from 'styled-components'
...
+import dimensions from '../constants/dimensions'
 import { textTruncator } from '../utils/stringUtils'
 import CleanLink from './CleanLink'
 import LastUpdateInfo from './LastUpdateInfo'
 import { Description } from './ListItem'
 
-export const NUM_OF_CHARS_ALLOWED = 220
+export const NUM_OF_CHARS_ALLOWED_LARGE_SCREEN = 220
+const NUM_OF_CHARS_ALLOWED_MEDIUM_SCREEN = 150
+const NUM_OF_CHARS_ALLOWED_SMALL_SCREEN = 100
```
Comment (klinzo) [inline web/src/components/NewsListItem.tsx:16]: Did you discuss these values with Nikola?

**Review by klinzo — CHANGES_REQUESTED** [review]
Please clean up the git history here :) Reset your local branch to master, cherry pick the relevant commits and force push.

Tested on phone and laptop on firefox

File: web/src/components/Tooltip.tsx (hunk 4 above)
Comment (SteffiStuffel) [inline web/src/components/Tooltip.tsx:4]:
```suggestion
import dimensions from '../constants/dimensions'
```

**Review by SteffiStuffel — COMMENTED** [review]
Not tested

**Review by klinzo — CHANGES_REQUESTED** [review]
May be bad UI/UX and should be rejected. Lets wait for some expertise here :)

**Comment by Cmd190** [comment]
Our Designer claimed that this is not neccessary

**Pull request closed without merging**